### PR TITLE
Improve ARM64 Silicon OS X and Ubuntu ARM64 support for VMWare Fusion

### PR DESCRIPTION
**Pull request by combatpoodle**
## Description
Fails to load the VM due to processor type mismatch.  Resolves with tacking on -arm64 or whatever onto the OS type in the template.

This initially starts with boot failures due to the LSILogic disk type preventing the VMX from loading at the vmxrun stage.  Changing the disk to SATA (longer term solution), or the disk adapter to PVSCSI from LSILOGIC resolves this.  There's builtin support for SATA, so I go that direction.

Next, IDE support for CDROM's is removed in Fusion 13.  Not sure if this is an M1/M2 thing or VMWare Fusion 13 change, I've been out of things for a bit.  Regardless, changing to SATA resolves it.  Just copied the same coding pattern from logical drives.

At this point things seem to boot to the point where they get into the GRUB loader then divert into PXE - so it's improved a good bit, but that's more of a software than virtualization issue.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I have run the pre-merge tests locally and they pass.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

_On checklist, existing tests do not appear to pass due to environment issues and are therefore not covered.  Tests are closer to passing now.  New tests are not necessary; integration (full machine build) tests should cover same functionality._

### CODE REVIEW

**Comment by israelshirk** [comment]
DCO is not reading correctly to me, I'm just tired ATM.  Will come back to it tomorrow or something.  Agree with the terms etc.  Thanks!

**Comment by combatpoodle** [comment]
Will run the linting/etc scripts in a bit.

**Comment by combatpoodle** [comment]
@Stromweld or others I think this is about as close as CI can get it?  Will try to follow up as soon as I have builds working-ish.

File: os_pkrvars/ubuntu/ubuntu-20.04-aarch64.pkrvars.hcl [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
+os_arch                   = "aarch64"
+iso_url                   = "http://cdimage.ubuntu.com/releases/20.04/release/ubuntu-20.04.5-live-server-arm64.iso"
+iso_checksum              = "sha256:e42d6373dd39173094af5c26cbf2497770426f42049f8b9ea3e60ce35bebdedf"
+hyperv_generation         = 2
+parallels_guest_os_type   = "ubuntu"
+vbox_guest_os_type        = "Ubuntu_64"
+vmware_guest_os_type      = "arm-ubuntu-64"
+boot_command              = ["<wait><esc>linux /casper/vmlinuz quiet autoinstall ds='nocloud-net;s=http://{{.HTTPIP}}:{{.HTTPPort}}/ubuntu/'<enter>initrd /casper/initrd<enter>boot<enter>"]
+vmware_cdrom_adapter_type = "sata"
+vmware_disk_adapter_type  = "sata"
```
Comment (Stromweld) [inline os_pkrvars/ubuntu/ubuntu-20.04-aarch64.pkrvars.hcl:12]: I'm thinking this makes sense and is the most compatible. Can you try making it the default for all vmware builds? If we set it at the variables level and then remove local variable in the pkr-sources file that it'd work for all builds and then we wouldn't need these extra variable overrides for all aarch64 builds.

File: packer_templates/pkr-variables.pkr.hcl [hunk 2]
```diff
@@ -175,9 +175,15 @@ variable "vbox_source" {
 }
 
 # vmware-iso
+variable "vmware_cdrom_adapter_type" {
+  type        = string
+  default     = null
+  description = "CDROM adapter type.  Needs to be SATA (or non-SCSI) for ARM64 builds."
+}
```
Comment (Stromweld) [inline packer_templates/pkr-variables.pkr.hcl:182]:
```suggestion
variable "vmware_cdrom_adapter_type" {
  type        = string
  default     = "sata"
  description = "CDROM adapter type.  Needs to be SATA (or non-SCSI) for ARM64 builds."
}
```

File: packer_templates/pkr-variables.pkr.hcl [hunk 3]
```diff
@@ -175,9 +175,15 @@ variable "vbox_source" {
...
 variable "vmware_disk_adapter_type" {
-  type    = string
-  default = null
+  type        = string
+  default     = null
```
Comment (Stromweld) [inline packer_templates/pkr-variables.pkr.hcl:185]:
```suggestion
  default     = "sata"
```

File: packer_templates/pkr-sources.pkr.hcl [hunk 4]
```diff
@@ -274,4 +274,5 @@ source "vmware-iso" "vm" {
   winrm_timeout                  = var.winrm_timeout
   winrm_username                 = var.winrm_username
   vm_name                        = local.vm_name
+  cdrom_adapter_type             = var.vmware_cdrom_adapter_type
```
Comment (Stromweld) [inline packer_templates/pkr-sources.pkr.hcl:277]: set the disk adaptor type to point to the var as well and then delete lines 67-70 on pkr-sources file.

**Review by Stromweld — CHANGES_REQUESTED** [review]

File: os_pkrvars/ubuntu/ubuntu-20.04-aarch64.pkrvars.hcl (hunk 1 above)
Comment (combatpoodle) [reply os_pkrvars/ubuntu/ubuntu-20.04-aarch64.pkrvars.hcl:12]: Agreed!  I was trying to minimize change scope but if you want repo-wide that’s definitely better/cleaner on all fronts.  My visibility is just a bit limited to make that call.

**Comment by combatpoodle** [comment]
Will throw in changes later tonight if I have time; added you as a collaborator on my repo @Stromweld if that’s a faster route to the finish line.  Thanks!

**Comment by Stromweld** [comment]
sure I'll give it a stab. That way they can then just build over night to see any results.

File: os_pkrvars/ubuntu/ubuntu-20.04-aarch64.pkrvars.hcl (hunk 1 above)
Comment (Stromweld) [reply os_pkrvars/ubuntu/ubuntu-20.04-aarch64.pkrvars.hcl]: worth testing

**Review by Stromweld — APPROVED** [review]
This looks good. I'm not seeing any changes in passing x86_64 machines for vmware. I forgot since I got the M2 mac mini aarch64 machines aren't able to initialize with the current vmware plugin.

**Pull request merged**